Switch license to MIT and update license files

karangattu opened the pull request:
Changed the package license from GPL-3 to MIT in DESCRIPTION, replaced the LICENSE file with a stub, added a full MIT LICENSE.md, and updated .Rbuildignore to exclude LICENSE.md. This aligns the package with the MIT license and clarifies licensing for users and contributors.

Closes #1170

karangattu committed:
Switch license to MIT and update license files

Changed the package license from GPL-3 to MIT in DESCRIPTION, replaced the LICENSE file with a stub, added a full MIT LICENSE.md, and updated .Rbuildignore to exclude LICENSE.md. This aligns the package with the MIT license and clarifies licensing for users and contributors.

gadenbuie commented on LICENSE:
```diff
@@ -1,303 +1,2 @@
-- jquery.highlight.js, https://github.com/bartaz/sandbox.js
-
-
-jQuery license
```
Rather than remove these licenses entirely, we should include a `LICENSE.note` file that includes them in full (details here: https://r-pkgs.org/license.html#sec-how-to-include). Would you mind creating that file?

gadenbuie replied:
Oh! Turns out leaflet is a good example: https://github.com/rstudio/leaflet/blob/relicense-GPL-to-MIT/LICENSE.note

karangattu replied:
Done! Take a look now

karangattu committed:
Add license notice for bundled third-party components

Introduces a LICENSE.note file listing third-party open source components bundled with the package and their respective license agreements. This improves transparency and compliance with license requirements for included dependencies.
